nixos/tests/buildbot.nix

```diff
@@ -4,43 +4,108 @@ import ./make-test.nix ({ pkgs, ... } : {
         masterUrl = "bbmaster:9989";
       };
+      environment.systemPackages = with pkgs; [ git buildbot-worker ];
+    };
+
+    gitrepo = { config, pkgs, ... }: {
+      services.openssh.enable = true;
+      networking.firewall.allowedTCPPorts = [ 22 9418 ];
+      environment.systemPackages = with pkgs; [ git ];
     };
   };
 
   testScript = ''
+    #Start up and populate fake repo
+    $gitrepo->waitForUnit("multi-user.target");
+    print($gitrepo->execute(" \
+      git config --global user.name 'Nobody Fakeuser' && \
+      git config --global user.email 'nobody\@fakerepo.com' && \
+      rm -rvf /srv/repos/fakerepo.git /tmp/fakerepo && \
+      mkdir -pv /srv/repos/fakerepo ~/.ssh && \
+      ssh-keyscan -H gitrepo > ~/.ssh/known_hosts && \
+      cat ~/.ssh/known_hosts && \
+      cd /srv/repos/fakerepo && \
+      git init && \
+      echo -e '#!/bin/sh\necho fakerepo' > fakerepo.sh && \
+      cat fakerepo.sh && \
+      touch .git/git-daemon-export-ok && \
+      git add fakerepo.sh .git/git-daemon-export-ok && \
+      git commit -m fakerepo && \
+      git daemon --verbose --export-all --base-path=/srv/repos --reuseaddr & \
+    "));
+
+    # Test gitrepo
+    $bbmaster->waitForUnit("network-online.target");
+    #$bbmaster->execute("nc -z gitrepo 9418");
+    print($bbmaster->execute(" \
+      rm -rfv /tmp/fakerepo && \
+      git clone git://gitrepo/fakerepo /tmp/fakerepo && \
+      pwd && \
+      ls -la && \
+      ls -la /tmp/fakerepo \
+    "));
 
-    $bbmaster->waitForUnit("network.target");
-    $bbworker->waitForUnit("network.target");
+    # Test start master and connect worker
+    $bbmaster->waitForUnit("buildbot-master.service");
+    $bbmaster->waitUntilSucceeds("curl -s --head http://bbmaster:8010") =~ /200 OK/;
+    $bbworker->waitForUnit("network-online.target");
+    $bbworker->execute("nc -z bbmaster 8010");
+    $bbworker->execute("nc -z bbmaster 9989");
+    $bbworker->waitForUnit("buildbot-worker.service");
+    print($bbworker->execute("ls -la /home/bbworker/worker"));
 
-    # Additional tests to be added
-    #$bbmaster->waitForUnit("buildbot-master.service");
-    #$bbmaster->waitUntilSucceeds("curl -s --head http://bbmaster:8010") =~ /200 OK/ or die;
-    #$bbworker->waitForUnit("buildbot-worker.service");
-    #$bbworker->waitUntilSucceeds("tail -10 /home/bbworker/worker/twistd.log") =~ /success/ or die;
+
+    # Test stop buildbot master and worker
+    print($bbmaster->execute(" \
+      systemctl -l --no-pager status buildbot-master && \
+      systemctl stop buildbot-master \
+    "));
+    $bbworker->fail("nc -z bbmaster 8010");
+    $bbworker->fail("nc -z bbmaster 9989");
+    print($bbworker->execute(" \
+      systemctl -l --no-pager status buildbot-worker && \
+      systemctl stop buildbot-worker && \
+      ls -la /home/bbworker/worker \
+    "));
+
+
+    # Test buildbot daemon mode
+    # NOTE: daemon mode tests disabled due to broken PYTHONPATH child inheritence
```

Comment:
Buildbot standalone daemon mode broken as of 49b20e5 and ca5a1d9 due to broken PYTHONPATH in `runCommand` `makeWrapper`.  Tests disabled.